Hyper/Collisions/SimulationManager.cs

```diff
@@ -0,0 +1,39 @@
+﻿using BepuPhysics;
+using BepuPhysics.CollisionDetection;
+using BepuUtilities;
+using BepuUtilities.Memory;
+
+namespace Hyper.Collisions;
+internal class SimulationManager<TNarrowPhaseCallbacks, TPoseIntegratorCallbacks> : IDisposable
+    where TNarrowPhaseCallbacks : struct, INarrowPhaseCallbacks
+    where TPoseIntegratorCallbacks : struct, IPoseIntegratorCallbacks
+{
+    public Simulation Simulation { get => _simulation; }
```

Comment:
why do we need _simulation? can't we just have get; private set;. same for BufferPool and ThreadDispatcher